collectors/metrics/pkg/status/status.go

```diff
@@ -93,3 +69,9 @@ func (s *StatusReport) UpdateStatus(ctx context.Context, reason status.Reason, m
 
 	return nil
 }
+
+type NoopReporter struct{}
```

Comment:
`forwarder.go` we use noop reporter by default - why the need for a status reporter at all there?

Reply:
Forwarder is in charge of federate and remote writing that federate. So that is where we report status.

Earlier it always reported status using kubeAPI, but for local testing might not need that. So I made the status reporting kind of optional.